hicommonwealth/commonwealth
# Truncate Bodies Script

## Link to Issue
Closes: #8671 

## Description of Changes
- Async migration to truncate bodies in the DB
- Modified routes to truncate bodies before inserting into the DB

## Test Plan
- (Required) get a fresh DB dump: `pnpm dump-db`
- Create a thread with more than 2k characters
- Execute `cd packages/commonwealth`
  - `pnpm ts-exec scripts/truncate-bodies.ts threads`
  - `pnpm ts-exec scripts/truncate-bodies.ts comments`
  - `pnpm ts-exec scripts/truncate-bodies.ts thread-versions`
  - `pnpm ts-exec scripts/truncate-bodies.ts comment-versions`
- View the thread you created before
- CRUD new threads with more than 2k characters and check version histories

## Deployment Plan

1. Execute the following after deployment:
```
NODE_ENV=production node --import=extensionless/register --max_old_space_size=$(./scripts/get-max-old-space-size.sh) --enable-source-maps ./build/scripts/truncate-bodies.js threads

NODE_ENV=production node --import=extensionless/register --max_old_space_size=$(./scripts/get-max-old-space-size.sh) --enable-source-maps ./build/scripts/truncate-bodies.js comments

NODE_ENV=production node --import=extensionless/register --max_old_space_size=$(./scripts/get-max-old-space-size.sh) --enable-source-maps ./build/scripts/truncate-bodies.js thread-versions

NODE_ENV=production node --import=extensionless/register --max_old_space_size=$(./scripts/get-max-old-space-size.sh) --enable-source-maps ./build/scripts/truncate-bodies.js comment-versions
```

## Other Considerations

-
